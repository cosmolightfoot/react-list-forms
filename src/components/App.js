import React, { useState } from 'react';
import Sidebar from './Sidebar';

export default function App() {
  const [selected, updateSelected] = useState('home');

  const content = {
    home: 'going home',
    about: 'yadayadayada me again',
    blog: 'still working on it',
    contact: 'NONE!'
  };

  return (
    <section>
      <Sidebar>
        <a onClick={() => updateSelected('home')} href="#">Home</a>
        <a onClick={() => updateSelected('about')} href="#">About</a>
        <a onClick={() => updateSelected('blog')} href="#">Blog</a>
        <a onClick={() => updateSelected('contact')} href="#">Contact</a>
      </Sidebar>
      <p>{content[selected]}</p>
    </section>
  );  
}
