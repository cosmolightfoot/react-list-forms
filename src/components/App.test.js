import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('app test', () => {
  it('app snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  }); 
});
