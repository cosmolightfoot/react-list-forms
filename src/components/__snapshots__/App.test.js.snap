// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app test app snapshot 1`] = `
<section>
  <Sidebar>
    <a
      href="#"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="#"
      onClick={[Function]}
    >
      About
    </a>
    <a
      href="#"
      onClick={[Function]}
    >
      Blog
    </a>
    <a
      href="#"
      onClick={[Function]}
    >
      Contact
    </a>
  </Sidebar>
  <p>
    going home
  </p>
</section>
`;
